src/aws/auth.rs

```diff
@@ -446,11 +481,44 @@ mod tests {
         match config.auth {
             AwsAuthentication::Role {
                 assume_role,
+                external_id,
+                load_timeout_secs,
+                imds,
+                region,
+            } => {
+                assert_eq!(&assume_role, "auth.root");
+                assert_eq!(external_id, None);
+                assert_eq!(load_timeout_secs, Some(10));
+                assert!(matches!(imds, ImdsAuthentication { .. }));
+                assert_eq!(region.unwrap(), "us-west-2");
+            }
+            _ => panic!(),
+        }
+    }
+
+    #[test]
+    fn parsing_both_external_id() {
+        let config = toml::from_str::<ComponentConfig>(
+            r#"
+            auth.assume_role = "auth.root"
+            external_id = "id"
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏ I'm curious about the test case, it looks like `external_id` should always be under `auth` - shouldn't it just be invalid to provide unnested?

Reply:
You are correct. It was more to the point that `auth.external_id` is the only source. May be a better test case would have been,  given `external_id` outside auth should assert it to be null.

Reply:
Gotcha, ideally we'd want to `[deny_unknown_fields]` to better surface to the user that they placed the option at the wrong depth.

Reply:
Thanks, since `#[serde(deny_unknown_fields)]` is already referenced, I have removed this unit test.